homeassistant/components/imgw_pib/sensor.py

```diff
@@ -0,0 +1,97 @@
+"""IMGW-PIB sensor platform."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from imgw_pib.model import HydrologicalData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfLength, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ImgwPibConfigEntry
+from .const import ATTRIBUTION
+from .coordinator import ImgwPibDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class ImgwPibSensorEntityDescription(SensorEntityDescription):
+    """IMGW-PIB sensor entity description."""
+
+    value: Callable[[HydrologicalData], StateType]
+
+
+SENSOR_TYPES: tuple[ImgwPibSensorEntityDescription, ...] = (
+    ImgwPibSensorEntityDescription(
+        key="water_level",
+        translation_key="water_level",
+        native_unit_of_measurement=UnitOfLength.CENTIMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=0,
+        value=lambda data: data.water_level.value,
+    ),
+    ImgwPibSensorEntityDescription(
+        key="water_temperature",
+        translation_key="water_temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value=lambda data: data.water_temperature.value,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ImgwPibConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a IMGW-PIB sensor entity from a config_entry."""
+    coordinator: ImgwPibDataUpdateCoordinator = entry.runtime_data.coordinator
```

Comment:
```suggestion
    coordinator = entry.runtime_data.coordinator
```
I think we can now do without